Add support for multicast ping

corny opened the pull request:
@stapelberg Please test and write your opinion. What do you think about the interface?

closes #4

stapelberg commented on sending.go:
```diff
@@ -30,15 +33,62 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
+	return req.roundTripTime()
+}
+
+// PingMulticast sends a single Echo Request and waits a given time for the answer(s)
```
Could you explain the channel semantics in the documentation here? I.e., the channel will be closed when the context is cancelled (I think?).

Also, maybe add an example to the documentation?
```go
ctx, canc := context.WithTimeout(context.Background(), 1 * time.Second)
defer canc()
responses, err := PingMulticast(ctx, net.ParseIP("ff02::1%eth0"))
if err != nil {
  return err
}
for resp := range responses {
  fmt.Println(resp)
}
```

stapelberg commented:
Thanks!

I ran the ping-test tool from the multicast branch and it works fine for my use-case (tried with ff02::1%eth0 and ff02::2%eth0).

I think the interface is fine, and left a comment about a possible documentation improvement.

corny commented:
Good idea - I will add one example.

Does `PingMulticast(ctx, net.ParseIP("ff02::1%eth0"))` really work? This should be correct:
```
addr, _ := ResolveIPAddr("ip", "ff02::1%eth0")
responses, err := PingMulticast(ctx, addr)
```

stapelberg commented:
> Does PingMulticast(ctx, net.ParseIP("ff02::1%eth0")) really work? This should be correct:

You are correct, `net.ParseIP()` is not sufficient, but your example does work :).

dmke commented on request.go:
```diff
@@ -1,20 +1,42 @@
+}
+
+// Response is a response to a multicast echo request
+type Response struct {
```
A response to an echo request is commonly known as "Reply" → `type Reply`?

corny replied:
Thanks, has been renamed.

dmke commented on request.go:
```diff
@@ -33,3 +64,31 @@ func (req *request) roundTripTime() (time.Duration, error) {
+
+// handleResponse is responsible for adding a result to the result set
+func (req *multiRequest) handleResponse(_ error, addr net.IP, tRecv *time.Time) {
+
```
Remove empty line.

dmke commented on sending.go:
```diff
@@ -19,7 +20,9 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
 		for i := 0; i < attempts; i++ {
-			if rtt, err = pinger.Ping(remote, timeout); err == nil {
+			rtt, err = pinger.Ping(remote, timeout)
+
```
Remove empty line.

dmke commented on sending.go:
```diff
@@ -30,15 +33,66 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
+	case <-ctx.Done():
+		// dequeue request
+		pinger.removeRequest(seq)
+
```
Remove empty line.

dmke commented on sending.go:
```diff
@@ -30,15 +33,66 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
+// The channel will be closed on termination of the context.
+// An error is returned if the sending of the echo request fails.
+func (pinger *Pinger) PingMulticast(ctx context.Context, remote *net.IPAddr) (<-chan Response, error) {
+
```
Remove empty line.

dmke commented on sending.go:
```diff
@@ -30,15 +33,66 @@ func (pinger *Pinger) PingAttempts(remote *net.IPAddr, timeout time.Duration, at
+		responses: make(chan Response),
+	}
+	seq, err := pinger.sendRequest(remote, &req)
+
```
Remove empty line.

corny commented:
I renamed `PingMulticast` to `PingMulticastContext` and added a `PingMulticast(destination *net.IPAddr, wait time.Duration)` method. I think passing a duration is more conventient than passing a context.
